src/scenes/ApplicationRouter/portal/MapUpload/index.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { Card, useTheme } from '@mui/material';
+import { BusySpinner } from '@terraware/web-components';
+import { FileChooser } from '@terraware/web-components';
+
+import { Crumb } from 'src/components/BreadCrumbs';
+import { APP_PATHS } from 'src/constants';
+import useNavigateTo from 'src/hooks/useNavigateTo';
+import { useLocalization } from 'src/providers';
+import { requestUploadApplicationBoundary } from 'src/redux/features/application/applicationAsyncThunks';
+import { selectApplicationUploadBoundary } from 'src/redux/features/application/applicationSelectors';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import ApplicationPage from 'src/scenes/ApplicationRouter/portal/ApplicationPage';
+import { useApplicationData } from 'src/scenes/ApplicationRouter/provider/Context';
+import strings from 'src/strings';
+import useSnackbar from 'src/utils/useSnackbar';
+
+const MapUploadView = () => {
+  const theme = useTheme();
+
+  const dispatch = useAppDispatch();
+  const { reload, selectedApplication } = useApplicationData();
+  const [requestId, setRequestId] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```

Comment:
Instead of creating more state, you can do something like `const isLoading = result?.status === 'pending'`